app/goal_routes.py

```diff
@@ -0,0 +1,97 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+from .routes_helper import validate_model
+
+goals_bp = Blueprint("goals", __name__, url_prefix = "/goals")
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response(jsonify({"details": "Invalid data"}), 400)
+    # Create a new goal in the database
+    new_goal = Goal(
+        title=request_body["title"]
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response_body = dict(goal = new_goal.to_dict())
+
+    return jsonify(response_body), 201
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+
+    goals = Goal.query.all()
+    goal_list = [goal.to_dict() for goal in goals]
+
+
+    return jsonify(goal_list), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model(Goal,goal_id)
+    response_body = dict(goal = goal.to_dict())
+
+    return jsonify(response_body), 200
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal,goal_id)
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
```

Comment:
Just like you have error handling in your POST route, you'll need error handling here (either using if or try/except) because there's a possibility that the client could send you a request body that doesn't have the key "title" or it's misspelled. If that were to happen then line 46 would throw an exception and without handling it, then the client would get a 500 internal server error in response.